fpdcc/ccfp-asset-dashboard
# Add DRF endpoint and serializers for LocalAssets model

## Overview
Adds DRF endpoints and serializers for saving local assets. There are no big UI changes, except for rendering a table in a django template. The table is not styled.

### Notes
Some UX things we need to improve:
1. Reload the search results when an asset is saved, like how the previous version worked when the search was managed by query string params. This version reloads the page and clears the search results since the search is happening via ajax calls. This could be a frustrating UX. I have a couple ideas how to do this.
2. When a user searches assets that overlap existing assets, the searched geometries show up on the map and not the existing geometries. This could be confusing -- a user needs to know what assets they already saved.
  - This is probably best explained by witnessing firsthand. To recreate:
    - Search "red" and select "Trails".
    - Save a section of those trails
    - The page reloads and shows the saved trails
    - Search "red" and select "Trails" again.
    - The searched geometries override the existing geometries.

Need to address several UX issues that are already tracked in Github issues.

Need to delete assets from the table. We could also add this functionality to the map -- it would be cool to select an asset in the map and delete from there...wouldn't be too hard to implement.

## Testing Instructions
- Visit review app: https://ccfp-asset-d-local-asse-klwnjy.herokuapp.com/projects/2/assets/
  - Login credentials in LastPass vault, "ccfp-asset-dashboard test user"
- Search for assets
- Select assets and save
  - Note that this map UX will change
- Make sure the geometries show up in the map and in the table
-  Save assets via the API
